Add sfs as valid system

WalterKolczynski-NOAA opened the pull request:
# Description
Adds sfs as a valid option for NET.

To start, the GEFS system is generally just copied wholesale for SFS. This includes the extract_vars job.

Other than base and resources, config files link to the GEFS versions, just as GEFS config files point to the GFS versions except where they have needed to be changed.

The temporary SFS_POST option has been removed.

The existing SFS test is copied and slightly modified for a PR-level CI test.

Resolves #2271

# Type of change
- [ ] Bug fix (fixes something broken)
- [x] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? YES
- Does this change require an update to any of the following submodules? NO

# How has this been tested?
- gfs cycled test on Hercules
- gefs test on Hercules

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [x] I have made corresponding changes to the system documentation if necessary

WalterKolczynski-NOAA committed:
Add sfs as valid system

Adds sfs as a valid option for NET.

To start, the GEFS system is generally just copied wholesale for SFS.
This includes the extract_vars job.

Other than base and resources, config files link to the GEFS versions,
just as GEFS config files point to the GFS versions except where they
have needed to be changed.

The temporarily SFS_POST option has been removed.

The sfs test case is moved from a separate sfs directory to the pr
directory, but ICs are still needed before it is functional.

Resolves #2271

WalterKolczynski-NOAA commented:
@NeilBarton-NOAA would you please supply ICs for the SFS test so we can set it up to run on all systems

WalterKolczynski-NOAA commented:
Still need to add SFS to the RTD documentation.

WalterKolczynski-NOAA committed:
Update docs for new sfs option

WalterKolczynski-NOAA committed:
Add missing sfs option to CICE postdet

WalterKolczynski-NOAA committed:
Restore old sfs case and make new one for PRs

The SFS case is moved back to its original position in an sfs directory,
then a copy is made in the PR directory using a shorter time period and
a couple other changes.

WalterKolczynski-NOAA commented:
ICs are in place and documentation has been updated. Ready for review.

XiaqiongZhou-NOAA commented:
@WenMeng-NOAA 

Wen noticed that three variables "u10m_max, v10m_max, wind10m_max" are missing in diag_table for SFS. These three variables are requested by CPC for SFS. Can this be added in diag_table?
It seems GFS/GEFS/SFS are using the same diag_table in workflow.

to be added in "diag_table" file as follows:
"gfs_phys",    "wind10mmax",    "wind10m_max",   "fv3_history2d",  "all",  .false.,  "none",  2
"gfs_phys",    "u10mmax",       "u10m_max",      "fv3_history2d",  "all",  .false.,  "none",  2
"gfs_phys",    "v10mmax",       "v10m_max",      "fv3_history2d",  "all",  .false.,  "none",  2

EricSinsky-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -564,28 +564,31 @@ FV3_predet(){
+        ${NCP} "${PARMgfs}/post/gefs/postxconfig-NT-gefs.txt"       "${DATA}/postxconfig-NT.txt"
+        ${NCP} "${PARMgfs}/post/gefs/postxconfig-NT-gefs-f00.txt"   "${DATA}/postxconfig-NT_FH00.txt"
+        # Provide ensemble header information for GEFS
+        if [[ "${ENSMEM}" == "000" ]]; then
+          export e1=1
+        else
+          export e1=3
+        fi
+        export e2="${ENSMEM:1:2}"
+        export e3="${NMEM_ENS}"        
```
I think ensemble header information can be added for the `sfs` case as well.

WalterKolczynski-NOAA committed:
Add ensemble grib header info for SFS

EricSinsky-NOAA commented on workflow/applications/sfs.py:
```diff
@@ -0,0 +1,98 @@
+            tasks += ['wavepostpnt']
+
+        if options['do_extractvars']:
+            tasks += ['extractvars', 'arch']
```
```suggestion
            tasks += ['extractvars']
```

EricSinsky-NOAA replied:
Should the `arch` task be created even when `do_extractvars` is `false`? This change has recently been applied to [gefs.py](https://github.com/NOAA-EMC/global-workflow/blob/develop/workflow/applications/gefs.py).

EricSinsky-NOAA commented on workflow/applications/sfs.py:
```diff
@@ -0,0 +1,98 @@
+        if options['do_extractvars']:
+            tasks += ['extractvars', 'arch']
+
+        tasks += ['cleanup']
```
```suggestion
        tasks += ['arch', 'cleanup']
```

EricSinsky-NOAA replied:
If the arch task were to be added unconditionally, will a `sfs_arcdir.yaml.j2 `and `master_sfs.yaml.j2` need to be created for the archive task to succeed?

WalterKolczynski-NOAA replied:
Turned off archival for now, but still going to load `config.archive`, since it doesn't hurt and is one fewer thing to add back later.

EricSinsky-NOAA commented on workflow/rocoto/sfs_tasks.py:
```diff
@@ -0,0 +1,632 @@
+        max_tasks = self._configs[config]['MAX_TASKS']
+        resources = self.get_resource(config)
+
+        fhrs = self._get_forecast_hours('gefs', self._configs[config], component)
```
```suggestion
        fhrs = self._get_forecast_hours('sfs', self._configs[config], component)
```

EricSinsky-NOAA commented on workflow/rocoto/sfs_tasks.py:
```diff
@@ -0,0 +1,632 @@
+
+        dependencies = rocoto.create_dependency(dep_condition='and', dep=deps)
+
+        fhrs = self._get_forecast_hours('gefs', self._configs['atmos_ensstat'])
```
```suggestion
        fhrs = self._get_forecast_hours('sfs', self._configs['atmos_ensstat'])
```

EricSinsky-NOAA commented on workflow/rocoto/sfs_tasks.py:
```diff
@@ -0,0 +1,632 @@
+        deps.append(rocoto.add_dependency(dep_dict))
+        dependencies = rocoto.create_dependency(dep=deps, dep_condition='or')
+
+        fhrs = self._get_forecast_hours('gefs', self._configs['wavepostsbs'], 'wave')
```
```suggestion
        fhrs = self._get_forecast_hours('sfs', self._configs['wavepostsbs'], 'wave')
```

EricSinsky-NOAA commented on parm/config/sfs/config.base:
```diff
@@ -0,0 +1,354 @@
+
+# Resolution specific parameters
+export LEVS=128
+export CASE="@CASECTL@"  # CASE is required in GEFS to determine ocean/ice/wave resolutions
```
`CASE` is also required for SFS, right? 

```suggestion
export CASE="@CASECTL@"  # CASE is required in SFS to determine ocean/ice/wave resolutions
```

EricSinsky-NOAA commented on parm/config/sfs/config.base:
```diff
@@ -0,0 +1,354 @@
+
+# Commonly defined parameters in JJOBS
+export envir=${envir:-"prod"}
+export NET="sfs"  # NET is defined in the job-card (ecf)
```
Since `NET` is set to `sfs`, [archive.py](https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/python/pygfs/task/archive.py) might fail at [lines 89-90](https://github.com/NOAA-EMC/global-workflow/blob/ee54eac82b6bf46ccbc6958697509fe2be441004/ush/python/pygfs/task/archive.py#L89) if the sfs yaml does not exist.

WalterKolczynski-NOAA replied:
Turned archive off for the time being, since the needs of SFS may be radically different, so just copying GEFS doesn't make sense.

EricSinsky-NOAA reviewed (changes requested):
Here are a couple of suggestions.

WalterKolczynski-NOAA committed:
Turn off SFS archival, add SFS fh computation

Adds SFS to the list of supported runs for `Tasks._get_forecast_hours()`

Turns off the archive job for SFS for the time being until an appropriate
yaml is created.

EricSinsky-NOAA reviewed (dismissed):
Looks good to me. Thank you for adding sfs as a valid option in gw.

WenMeng-NOAA commented:
@WalterKolczynski-NOAA Would you like to include sfs-specified atmos. product generation (pgrb2 files) in this PR? If so, I can provide the product list.

WalterKolczynski-NOAA commented:
> @WalterKolczynski-NOAA Would you like to include sfs-specified atmos. product generation (pgrb2 files) in this PR? If so, I can provide the product list.

@WenMeng-NOAA Is it just linking in new files from UPP, or is there additional work?

WenMeng-NOAA commented:
> > @WalterKolczynski-NOAA Would you like to include sfs-specified atmos. product generation (pgrb2 files) in this PR? If so, I can provide the product list.
> 
> @WenMeng-NOAA Is it just linking in new files from UPP, or is there additional work?

@WalterKolczynski-NOAA The product list is sfs specified, similar to [gfs product list](https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/product/gefs.1p00.f000.paramlist.a.txt). I created this list based on the UPP output for sfs and would suggest the product generation process only generates one dataset "pgrb2" in 1p00 deg. instead of two datasets "pgrb2" and "pgrb2b". There might be additional work for you to update relevant scripts for sfs.

WalterKolczynski-NOAA commented:
> > > @WalterKolczynski-NOAA Would you like to include sfs-specified atmos. product generation (pgrb2 files) in this PR? If so, I can provide the product list.
> > 
> > 
> > @WenMeng-NOAA Is it just linking in new files from UPP, or is there additional work?
> 
> @WalterKolczynski-NOAA The product list is sfs specified, similar to [gfs product list](https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/product/gefs.1p00.f000.paramlist.a.txt). I created this list based on the UPP output for sfs and would suggest the product generation process only generates one dataset "pgrb2" in 1p00 deg. instead of two datasets "pgrb2" and "pgrb2b". There might be additional work for you to update relevant scripts for sfs.

I'm going to leave that for a future PR then. This is just to get the SFS framework in.

WenMeng-NOAA commented:
> > > > @WalterKolczynski-NOAA Would you like to include sfs-specified atmos. product generation (pgrb2 files) in this PR? If so, I can provide the product list.
> > > 
> > > 
> > > @WenMeng-NOAA Is it just linking in new files from UPP, or is there additional work?
> > 
> > 
> > @WalterKolczynski-NOAA The product list is sfs specified, similar to [gfs product list](https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/product/gefs.1p00.f000.paramlist.a.txt). I created this list based on the UPP output for sfs and would suggest the product generation process only generates one dataset "pgrb2" in 1p00 deg. instead of two datasets "pgrb2" and "pgrb2b". There might be additional work for you to update relevant scripts for sfs.
> 
> I'm going to leave that for a future PR then. This is just to get the SFS framework in.

@WalterKolczynski-NOAA Sure, your plan works for me.

aerorahul commented on ush/forecast_postdet.sh:
```diff
@@ -536,52 +542,57 @@ MOM6_postdet() {
-        continue
-      fi
+  case ${RUN} in
+    *gfs|gefs|sfs) # Link output files for RUN=gfs|enkfgfs|gefs|sfs
```
```suggestion
    gfs|enkfgfs|gefs|sfs) # Link output files for RUN=gfs|enkfgfs|gefs|sfs
```
Making this explicit would also make it clear.  Thoughts?

WalterKolczynski-NOAA replied:
I'm fine with being explicit

WalterKolczynski-NOAA replied:
Done

aerorahul commented on ush/forecast_postdet.sh:
```diff
@@ -536,52 +542,57 @@ MOM6_postdet() {
-      ${NLN} "${COMOUT_OCEAN_HISTORY}/${RUN}.ocean.t${cyc}z.inst.f${fhr3}.nc" "${DATA}/MOM6_OUTPUT/ocn_da_${vdatestr}.nc"
-    done
-  fi
+    *gdas) # Link output files for RUN=gdas|enkfgdas
```
```suggestion
    gdas|enkfgdas) # Link output files for RUN=gdas|enkfgdas
```
same comment

WalterKolczynski-NOAA committed:
Make RUNs explicit in postdet

aerorahul commented on parm/config/sfs/yaml/defaults.yaml:
```diff
@@ -0,0 +1,34 @@
+fcst:
+  reforecast: "NO"
+  FHZER: 6
+  TYPE: "nh"
+  MONO: "non-mono"
```
Where are these templated?

WalterKolczynski-NOAA replied:
`config.fcst` (gets used in predet)

aerorahul commented on ush/forecast_postdet.sh:
```diff
@@ -305,18 +305,24 @@ FV3_out() {
-    fi
-  fi
+  case ${RUN} in
+    gdas|enkf*) # Copy restarts in the assimilation window for RUN=gdas|enkfgdas|enkfgfs
```
```suggestion
    gdas|enkfgdas|enkfgfs) # Copy restarts in the assimilation window for RUN=gdas|enkfgdas|enkfgfs
```

aerorahul commented on ush/forecast_postdet.sh:
```diff
@@ -614,30 +625,33 @@ MOM6_out() {
-    if [[ "${RUN}" == "gfs" || "${RUN}" == "gefs" ]]; then
-      echo "Copying MOM6 restarts for 'RUN=${RUN}' at ${forecast_end_cycle}"
+  case ${RUN} in
+    gdas|enkf*) # Copy restarts for the next cycle for RUN=gdas|enkfgdas|enkfgfs
```
```suggestion
    gdas|enkfgdas|enkfgfs) # Copy restarts for the next cycle for RUN=gdas|enkfgdas|enkfgfs
```

aerorahul commented on ush/forecast_postdet.sh:
```diff
@@ -716,31 +737,31 @@ CICE_out() {
-    local seconds source_file target_file
-    if [[ "${RUN}" == "gfs" || "${RUN}" == "gefs" ]]; then
+  case ${RUN} in
+    gdas|enkf*) # Copy restarts for next cycle for RUN=gdas|enkfgdas|enkfgfs
```
```suggestion
    gdas|enkfgdas|enkfgfs) # Copy restarts for next cycle for RUN=gdas|enkfgdas|enkfgfs
```

aerorahul commented on ush/forecast_postdet.sh:
```diff
@@ -846,35 +867,39 @@ CMEPS_postdet() {
-  fi
-
+  case ${RUN} in
+    gdas|enkf*) # Copy restarts for the next cycle to COM for RUN=gdas|enkfgdas|enkfgfs
```
```suggestion
    gdas|enkfgdas|enkfgfs) # Copy restarts for the next cycle to COM for RUN=gdas|enkfgdas|enkfgfs
```

aerorahul commented on ush/forecast_predet.sh:
```diff
@@ -564,28 +564,39 @@ FV3_predet(){
+        export e3="${NMEM_ENS}"
+        ;;
+      *)
+        echo "FATAL ERROR: Unknown RUN ${RUN}, unable to determine appropriate post files"
```
```suggestion
        echo "FATAL ERROR: Unknown NET ${NET}, unable to determine appropriate post files"
```

aerorahul commented on parm/config/gefs/config.ufs:
```diff
@@ -326,11 +326,16 @@ if [[ "${skip_mom6}" == "false" ]]; then
+      case ${RUN} in
+        gfs|gefs|sfs)
+          MOM6_DIAG_MISVAL="-1e34";;
+        gdas|enkf*)
```
```suggestion
        gdas|enkfgdas|enkfgfs)
```
Not that any of these will be executed in the `NET=gefs` system.

aerorahul commented on parm/config/gefs/config.ufs:
```diff
@@ -349,13 +354,20 @@ if [[ "${skip_mom6}" == "false" ]]; then
+          MOM6_DIAG_COORD_DEF_Z_FILE="interpolate_zgrid_40L.nc"
+          MOM6_DIAG_MISVAL="-1e34"
+          ;;
+        gdas|enkf*)
```
```suggestion
        gdas|enkfgdas|enkfgfs)
```

aerorahul commented on parm/config/gefs/config.ufs:
```diff
@@ -370,13 +382,20 @@ if [[ "${skip_mom6}" == "false" ]]; then
+          MOM6_DIAG_COORD_DEF_Z_FILE="interpolate_zgrid_40L.nc"
+          MOM6_DIAG_MISVAL="-1e34"
+          ;;
+        gdas|enkf*)
```
```suggestion
        gdas|enkfgdas|enkfgfs)
```

aerorahul commented on parm/config/gefs/config.ufs:
```diff
@@ -392,13 +411,20 @@ if [[ "${skip_mom6}" == "false" ]]; then
+          MOM6_DIAG_COORD_DEF_Z_FILE="interpolate_zgrid_40L.nc"
+          MOM6_DIAG_MISVAL="-1e34"
+          ;;
+        gdas|enkf*)
```
```suggestion
        gdas|enkfgdas|enkfgfs)
```

aerorahul commented on parm/config/gfs/config.ufs:
```diff
@@ -436,11 +436,16 @@ if [[ "${skip_mom6}" == "false" ]]; then
+      case ${RUN} in
+        gfs|gefs|sfs)
+          MOM6_DIAG_MISVAL="-1e34";;
+        gdas|enkf*)
```
```suggestion
        gdas|enkfgdas|enkfgfs)
```

aerorahul commented on parm/config/gfs/config.ufs:
```diff
@@ -459,13 +464,20 @@ if [[ "${skip_mom6}" == "false" ]]; then
+          MOM6_DIAG_COORD_DEF_Z_FILE="interpolate_zgrid_40L.nc"
+          MOM6_DIAG_MISVAL="-1e34"
+          ;;
+        gdas|enkf*)
```
```suggestion
        gdas|enkfgdas|enkfgfs)
```

aerorahul commented on parm/config/gfs/config.ufs:
```diff
@@ -480,13 +492,20 @@ if [[ "${skip_mom6}" == "false" ]]; then
+          MOM6_DIAG_COORD_DEF_Z_FILE="interpolate_zgrid_40L.nc"
+          MOM6_DIAG_MISVAL="-1e34"
+          ;;
+        gdas|enkf*)
```
```suggestion
        gdas|enkfgdas|enkfgfs)
```

aerorahul commented on parm/config/gfs/config.ufs:
```diff
@@ -502,13 +521,20 @@ if [[ "${skip_mom6}" == "false" ]]; then
+          MOM6_DIAG_COORD_DEF_Z_FILE="interpolate_zgrid_40L.nc"
+          MOM6_DIAG_MISVAL="-1e34"
+          ;;
+        gdas|enkf*)
```
```suggestion
        gdas|enkfgdas|enkfgfs)
```

WalterKolczynski-NOAA committed:
Making more cases explicit

DavidHuber-NOAA commented on workflow/rocoto/sfs_tasks.py:
```diff
@@ -0,0 +1,632 @@
+    def waveinit(self):
+
+        resources = self.get_resource('waveinit')
+        task_name = f'{self.run}_wave_init'
```
```suggestion
        task_name = f'{self.run}_waveinit'
```

WalterKolczynski-NOAA replied:
`sfs_tasks.py` is literally a direct copy of `gefs_tasks.py` with the class name changed.

DavidHuber-NOAA replied:
Works for me.  Thanks.

DavidHuber-NOAA commented on workflow/rocoto/sfs_tasks.py:
```diff
@@ -0,0 +1,632 @@
+        dependencies.append(rocoto.add_dependency(dep_dict))
+
+        if self.options['do_wave']:
+            dep_dict = {'type': 'task', 'name': f'{self.run}_wave_init'}
```
```suggestion
            dep_dict = {'type': 'task', 'name': f'{self.run}_waveinit'}
```

DavidHuber-NOAA replied:
Looking at this more, I think this is by design.  Is that right, @WalterKolczynski-NOAA?

DavidHuber-NOAA replied:
Resolving.

DavidHuber-NOAA commented on workflow/rocoto/sfs_tasks.py:
```diff
@@ -0,0 +1,632 @@
+                         'dependency': dependencies,
+                         'envars': efcsenvars,
+                         'cycledef': f'{self.run}',
+                         'command': f'{self.HOMEgfs}/jobs/rocoto/fcst.sh',
```
Should this be efcs.sh?

aerorahul reviewed (dismissed):
looks ok to me.

WalterKolczynski-NOAA committed:
Merge branch 'develop' into feature/sfs

WalterKolczynski-NOAA commented:
CI failed quietly, but also seems unrelated to the PR. Probably a few separate issues going on.

WalterKolczynski-NOAA committed:
Merge branch 'develop' into feature/sfs

TerrenceMcGuinness-NOAA commented:
**C96mx100_S2S** fails to get `gefs/config.arch` during `setup_expt.py` due to a malformed soft link:

```
mterry (hercules-login-4) workflow ((HEAD detached at 86b758ec)) $ pwd
/work2/noaa/global/CI/HERCULES/3243/global-workflow/workflow

mterry (hercules-login-4) workflow ((HEAD detached at 86b758ec)) $ ./setup_expt.py sfs forecast-only --idate 1994050100 --edate 1994050100 --pslot C96mx100_S2S_86b758ec --app S2S --resdetatmos 96 --resensatmos 96 --resdetocean 1 --start cold --nens 2 --comroot /work2/noaa/global/CI/HERCULES/3243/COMROOT --expdir /work2/noaa/global/CI/HERCULES/3243/EXPDIR --icsdir /work/noaa/global/glopara/data/ICSDIR/C96mx100/20240610 --yaml /work2/noaa/global/CI/HERCULES/3243/global-workflow/ci/cases/yamls/sfs_defaults.yaml --overwrite

directory already exists in /work2/noaa/global/CI/HERCULES/3243/COMROOT/C96mx100_S2S_86b758ec
removing directory ........ /work2/noaa/global/CI/HERCULES/3243/COMROOT/C96mx100_S2S_86b758ec


directory already exists in /work2/noaa/global/CI/HERCULES/3243/EXPDIR/C96mx100_S2S_86b758ec
removing directory ........ /work2/noaa/global/CI/HERCULES/3243/EXPDIR/C96mx100_S2S_86b758ec

Traceback (most recent call last):
  File "/work2/noaa/global/CI/HERCULES/3243/global-workflow/workflow/./setup_expt.py", line 454, in <module>
    main()
  File "/work2/noaa/global/CI/HERCULES/3243/global-workflow/workflow/./setup_expt.py", line 443, in main
    fill_expdir(user_inputs)
  File "/work2/noaa/global/CI/HERCULES/3243/global-workflow/workflow/./setup_expt.py", line 44, in fill_expdir
    shutil.copy(config, expdir)
  File "/work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/envs/gsi-addon-env/install/intel/2021.9.0/python-3.11.6-ajcuas2/lib/python3.11/shutil.py", line 419, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/envs/gsi-addon-env/install/intel/2021.9.0/python-3.11.6-ajcuas2/lib/python3.11/shutil.py", line 256, in copyfile
    with open(src, 'rb') as fsrc:
         ^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/work2/noaa/global/CI/HERCULES/3243/global-workflow/parm/config/sfs/config.arch'

mterry (hercules-login-4) workflow ((HEAD detached at 86b758ec)) $ ls -l /work2/noaa/global/CI/HERCULES/3243/global-workflow/parm/config/sfs/config.arch

lrwxrwxrwx 1 mterry global 19 Feb  5 09:52 /work2/noaa/global/CI/HERCULES/3243/global-workflow/parm/config/sfs/config.arch -> ../gefs/config.arch

mterry (hercules-login-4) workflow ((HEAD detached at 86b758ec)) $ ls ../gefs/config.arch
ls: cannot access '../gefs/config.arch': No such file or directory

```

WalterKolczynski-NOAA committed:
Fix symlinks for sfs archive configs

During the reconcilliation with the merged PR that split the archive
job, the SFS config symlinks for the archive config file was not
updated.

DavidHuber-NOAA commented on parm/config/sfs/config.resources:
```diff
@@ -0,0 +1,363 @@
+    export is_exclusive=False
+    ;;
+
+  "arch")
+    export walltime="06:00:00"
+    export ntasks=1
+    export tasks_per_node=1
+    export threads_per_task=1
+    export memory="4096M"
+    ;;
```
Should have two jobs sets of resources here.  One for `arch_tars` and one for `arch_vrfy`.

WalterKolczynski-NOAA replied:
Fixed, along with similar reconciliation issues in config.base

WalterKolczynski-NOAA committed:
Reconcile additional config issues

WalterKolczynski-NOAA committed:
Fix SFS cleanup deps

WalterKolczynski-NOAA committed:
Remove replay test from CI

WalterKolczynski-NOAA committed:
Restore COPY_FINAL_RESTARTS flag for ice

The flag for copying the final restarts was inadvertently removed for
ice during refactoring.

WalterKolczynski-NOAA commented:
Will look closer tomorrow, but I think cleanup may be triggering before wave post point is done.

WalterKolczynski-NOAA committed:
Merge branch 'develop' into feature/sfs

WalterKolczynski-NOAA commented:
Looks like `cleanup` ran before `verfrad` completed. I think there are still some missing dependences there resulting from #3242.

WalterKolczynski-NOAA committed:
Merge branch 'develop' into feature/sfs

DavidHuber-NOAA commented on parm/config/sfs/config.resources:
```diff
@@ -0,0 +1,371 @@
+
+echo "BEGIN: config.resources"
+
+case ${machine} in
+  "WCOSS2")   max_tasks_per_node=128;;
+  "HERA")     max_tasks_per_node=40;;
+  "ORION")    max_tasks_per_node=40;;
+  "HERCULES") max_tasks_per_node=80;;
+  "JET")
+    case ${PARTITION_BATCH} in
+      "xjet")          max_tasks_per_node=24;;
+      "vjet" | "sjet") max_tasks_per_node=16;;
+      "kjet")          max_tasks_per_node=40;;
+      *)
+        echo "FATAL ERROR: Unknown partition ${PARTITION_BATCH} specified for ${machine}"
+        exit 3
+    esac
+    ;;
+  "S4")
+    case ${PARTITION_BATCH} in
+      "s4")  max_tasks_per_node=32;;
+      "ivy") max_tasks_per_node=20;;
+      *)
+        echo "FATAL ERROR: Unknown partition ${PARTITION_BATCH} specified for ${machine}"
+        exit 3
+    esac
+    ;;
+  "AWSPW")
+    export PARTITION_BATCH="compute"
+    max_tasks_per_node=48
+    ;;
+  "AZUREPW")
+    export PARTITION_BATCH="compute"
+    max_tasks_per_node=36
```
A thought for another time: it might be nice to have a common resources file for machine parameters and tasks that run for all `NET`s.

DavidHuber-NOAA replied:
Opened issue https://github.com/NOAA-EMC/global-workflow/issues/3336 to look into this.  Resolving this conversation.

DavidHuber-NOAA reviewed (approved):
Looks good!  One comment for a later time.  I'll make a follow-up issue.
